src/apis/pypi.py

```diff
@@ -9,4 +15,61 @@ def is_package_on_pypi(package_name, version=None):
     elif response.status_code == 404:
         return False
     else:
-        return None
\ No newline at end of file
+        return None
+
+
+def get_latest_package_version(package_name):
+    url = f'https://pypi.org/pypi/{package_name}/json'
+    response = requests.get(url)
+    if response.status_code != 200:
+        return None
+    data = response.json()
+    releases = data['releases']
+
+    # Get package versions not older than 2021
+    valid_versions = []
+    for v, release_info in releases.items():
+        upload_time = datetime.strptime(release_info[0]['upload_time'], '%Y-%m-%dT%H:%M:%S')
+        if upload_time.year <= 2021:
+            valid_versions.append(v)
+
+    v = max(valid_versions, key=version.parse) if valid_versions else None
+    return v
+
+
+def clean_requirements_txt(previous_microservice_path):
```

Comment:
Can you add documentation what does this function does?

Reply:
done thanks